featuretools/computational_backends/pandas_backend.py

```diff
@@ -245,15 +260,26 @@ def _calculate_features_for_entity(self, entity_id, feature_trie, df_trie,
             else:
                 sub_entity = relationship.child_entity.id
                 sub_filter_variable = relationship.child_variable.id
-                sub_filter_values = df[relationship.parent_variable.id]
-                parent_data = (relationship, new_ancestor_relationship_variables, df)
+
+                # Pass filtered values, even if we are using a full df.
+                if full_entity_features:
+                    filtered_df = df[df[filter_variable].isin(filter_values)]
+                else:
+                    filtered_df = df
+                sub_filter_values = filtered_df[relationship.parent_variable.id]
+
+                parent_data = (relationship,
+                               new_ancestor_relationship_variables,
+                               df)
```

Comment:
can parent data get away with just the `filtered_df`? i'd think this okay because of `sub_filter_values `is based on the `filtered_df`?

Reply:
My thinking was that if the child node needs the full entity then it will need the full `parent_data` to get the relationship variables. But no tests fail if we pass `filtered_df` instead, so either I'm wrong or we don't have a test for such a case.

Reply:
ah, i see. even though we pass the `filtered_df` through, if could be ignored if a child feature uses full entity. let's leave as-is. ideally, write a test

Reply:
actually, the more i think about it, I can't think of what feature the child entity could define on the full entity that would rely on a ancestor relationship variable. maybe see if you can write a test and if not, just use the filtered df

Reply:
Here is the test: https://github.com/Featuretools/featuretools/pull/598/files#diff-bf66dc93fe293a61ccc3fbd3c92eec31R76

The ancestor relationship variable is necessary when we calculate the agg. If this is changed to `filtered_df` then the test fails because the aggregation is `0` for the rows of the target other than the `instance_ids`. The rows of `log` that should be linked to those rows of the target entity have `NaN` for the relationship variable. Does that make sense?

Reply:
i see. thanks